File: jetbrains-plugin/src/main/kotlin/com/ezevillo/gitreview/domain/EntryArg.kt
package com.ezevillo.gitreview.domain

/**
 * Resolves the entry a command should operate on: explicit [EntryRecord], or
 * current by position when arg is null.
 */
fun resolveEntryArg(
    arg: Any?,
    entries: List<EntryRecord>,
    position: Int?,
): EntryRecord? {
    if (arg == null) {
        return entries.find { it.position == position }
    }
    if (arg is EntryRecord) {
        return arg
    }
    return null
}

File: visualstudio-extension/src/GitReview.Domain/EntryArg.cs
namespace GitReview.Domain;

public static class EntryArg
{
    public static EntryRecord? ResolveEntryArg(
        object? arg,
        IReadOnlyList<EntryRecord> entries,
        int? position)
    {
        if (arg is null)
            return entries.FirstOrDefault(e => e.Position == position);
        if (arg is EntryRecord er)
            return er;
        return null;
    }
}
